### ISSUE TITLE
Publish mqtt message once only

### ISSUE THREAD

**Issue by jlq76** [ronschaeffer/sbm2mqtt#4]
Hi, 

I noticed the script would publish only once on my mqtt broker despite no error log.
I suspect it's coming from the `mqttc.publish()` function.

I did the following change and it seems to fix the issue so far (up and running for 2 days every 5 minutes with two meters)

import mqtt.publish instead of mqtt.client:
```
import paho.mqtt.publish as publish 
```

replaced the call to `mqttc.publish()` by `publish.single()`.
Reason: it closes the connection in a clean way after publishing
[https://www.eclipse.org/paho/clients/python/docs/#single](url)

```
#mqttc = mqtt.Client(mqtt_client) 
#mqttc.username_pw_set(mqtt_user, mqtt_pass)
#mqttc.connect(mqtt_host, mqtt_port)
#result, mid = mqttc.publish(mqtt_topic + mac, msg_data)
publish.single(mqtt_topic + mac, msg_data, qos=0, hostname=mqtt_host, port=mqtt_port, client_id=mqtt_client)
```

I hope it helps.
/JL

**Comment by ronschaeffer** [comment]
Thanks. That sounds like a better solution. I'll test on my own system and then update if all goes well. Feel free to make a pull request if you'd like.

**Comment by ronschaeffer** [comment]
I finally got around to doing a quick test. @jlq76 . I haven't tried to work out why yet, but this hangs indefinitely for me after publishing data for one of two Meters. Complete code:

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# sbm2mqtt.py https://github.com/ronschaeffer/sbm2mqtt
#
# Based in part on:
#   https://github.com/OpenWonderLabs/python-host
#   https://github.com/bbostock/Switchbot_Py_Meter
#   https://qiita.com/warpzone/items/11ec9bef21f5b965bce3.

from bluepy.btle import Scanner, DefaultDelegate
import datetime
import binascii
import paho.mqtt.publish as publish
 
# Import configuration variables from sbm2mqtt_config.py file - Must be in the same folder as this script
from sbm2mqtt_config import (
    mqtt_host,
    mqtt_port,
    mqtt_timeout,
    mqtt_client,
    mqtt_user,
    mqtt_pass,
    mqtt_topic,
)

# SwitchBot UUID - See https://github.com/OpenWonderLabs/python-host/wiki/Meter-BLE-open-API
service_uuid = "cba20d00-224d-11e6-9fb8-0002a5d5c51b"


class ScanDelegate(DefaultDelegate):
    def __init__(self):
        DefaultDelegate.__init__(self)

    # Scan for BLE device advertisements, filter out ones which are not SwitchBot Meters & convert the service data to binary
    def handleDiscovery(self, dev, isNewDev, isNewData):
        for (adtype, desc, value) in dev.getScanData():
            if (
                adtype == 7 and value == service_uuid
            ):  # Check for devices with Switchnot devices
                mac = dev.addr
                for (adtype, desc, value) in dev.getScanData():
                    if (
                        len(value) == 16 and value[4:6] == "54"
                    ):  # Check for model "T" (54) in 16b service data
                        binvalue = binascii.unhexlify(
                            value
                        )  # Convert service data from hex to binary; See BLE API docs and warpzone link above for value mapping

                        # Get temperature and related characteristics
                        temperature = (binvalue[6] & 0b01111111) + (
                            (binvalue[5] & 0b00001111) / 10
                        )  # Absolute value of temp
                        if not (binvalue[6] & 0b10000000):  # Is temp negative?
                            temperature = -temperature
                        if not (binvalue[7] & 0b10000000):  # C or F?
                            temp_scale = "C"
                        else:
                            temp_scale = "F"
                            temperature = round(
                                temperature * 1.8 + 32, 1
                            )  # Convert to F

                        # Get other info
                        humidity = binvalue[7] & 0b01111111
                        battery = binvalue[4] & 0b01111111

                        # Get current time
                        time = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")

                        # Print info to terminal
                        print("\n" + mac + " @ " + str(time))
                        print("  Temp: " + str(temperature) + "\u00B0" + temp_scale)
                        print("  Humidity: " + str(humidity) + "%")
                        print("  Battery: " + str(battery) + "%")

                        # MQTT publish as JSON
                        msg_data = (
                            '{"time":"'
                            + time
                            + '","temperature":'
                            + str(temperature)
                            + ',"humidity":'
                            + str(humidity)
                            + ',"battery":'
                            + str(battery)
                            + ',"temperature_scale":"'
                            + temp_scale
                            + '"}'
                        )
                        print(
                            "\n  Publishing MQTT payload to "
                            + mqtt_topic
                            + mac
                            + " ...\n\n    "
                            + msg_data
                        )
                        #mqttc = mqtt.Client(mqtt_client)
                        #mqttc.username_pw_set(mqtt_user, mqtt_pass)
                        #mqttc.connect(mqtt_host, mqtt_port)
                        #mqttc.publish(mqtt_topic + mac, msg_data, 1)
                        publish.single(mqtt_topic + mac, msg_data, qos=0, hostname=mqtt_host, port=mqtt_port, client_id=mqtt_client)


def main():

    print("\nScanning for SwitchBot Meters...")
    scan = scanner = Scanner().withDelegate(ScanDelegate())
    scanner.scan(10.0)

    print("\nFinished.\n")


if __name__ == "__main__":
    main()

```
Hangs indefinitely here after publish data from the first Meter:
![image](https://user-images.githubusercontent.com/12379923/85885734-bf655a00-b7dc-11ea-8929-03594f2a15ca.png)

**Comment by jlq76** [comment]
Sorry for the late reply.
So far, the modified version, using `publish.single()` is still working on my side.
It has been up and running for more than a month.

The only issue I encountered since is unrelated and is with the bluetooth:
`bluepy.btle.BTLEManagementError: Failed to execute management command 'scanend' (code: 11, error: Rejected)`

I did double check the code again and this is the only change I made.
I also noticed that I kept` import paho.mqtt.client as mqtt` uncommented but commenting it out doesn't change anything

Maybe the different behavior is caused by the MQTT broker?
I'm using broker.hivemq.com
